src/Tribe/Aggregator/Record/Queue_Processor.php

```diff
@@ -164,11 +178,23 @@ public function set_current_queue( Tribe__Events__Aggregator__Record__Queue_Inte
 	 *
 	 * If no records in need of further processing can be found it will return bool false.
 	 *
+	 * @since TBD Inclusion of a $cache param for performance purposes.
+	 *
 	 * @param boolean $interactive_only Whether or not we should look for imports that were kicked off interactively
+	 * @param boolean $cache            When checking on every request we should have a transient caching to prevent hitting the DB every time.
 	 *
-	 * @return boolean
+	 * @return boolean|integer
 	 */
-	public function next_waiting_record( $interactive_only = false ) {
+	public function next_waiting_record( $interactive_only = false, $cache = false ) {
+		if ( true === $cache ) {
+			$transient_key = 'tribe-event-aggregator-next_waiting_record' . ( ! $interactive_only ?: '_interactive_only' );
```

Comment:
* :warning: Equals sign not aligned with surrounding assignments; expected 7 spaces but found 1 space
* :no_entry_sign: Using short ternaries is not allowed

<sub>hash: 8e5ecf9582200c347dba8b7ac0cdb4d1</sub>